### PR TITLE
Bug 1144558 - Include build system type in job list / job output r=edmor...

### PR DESCRIPTION
**Pull request by lightsofapollo**
...ley

### CODE REVIEW

File: treeherder/model/derived/jobs.py [hunk 1]
```diff
@@ -300,7 +300,7 @@ def get_job_list(self, offset, limit,
             replace=repl,
             placeholders=placeholders,
             limit="{0},{1}".format(offset, limit),
-            debug_show=self.DEBUG,
+            debug_show=True,
```
Comment (edmorley) [inline treeherder/model/derived/jobs.py]: Leftover from debugging?

**Comment by edmorley** [comment]
JOB_PROPERTIES also needs updating:
https://github.com/mozilla/treeherder-service/blob/70330c5842a51d06a81a2f1f60cc1b1988c0ff02/treeherder/webapp/api/utils.py#L23

It's mildly concerning the tests passed without this..

**Comment by edmorley** [comment]
Actually, Mauro's just spotted that we no longer use JOB_PROPERTY_RETURN_KEY and thus JOB_PROPERTIES, after bug 1097090. I'll clean that up separately.

If you could remove the debug_show change, this should be good to land - thanks! :-)

**Comment by maurodoglio** [comment]
One thing I would add to this patch is the ability to filter on build_system_type. To do that you just need to  include a mapping for it in https://github.com/mozilla/treeherder-service/blob/master/treeherder/model/derived/jobs.py#L102-L122

**Comment by lightsofapollo** [comment]
Awesome thanks! Will push if this green (I addressed comments...)

**Pull request merged**